### PR TITLE
Refactor networking

### PR DESCRIPTION
**Pull request by pacjac**
Mostly directory structure and convenience refactoring.

1. Implemented Publisher and Subscriber abstract base class for the sake of type hints.
2. Moved mqtt, zmq and serial wrappers to msb/networking to not clutter msb/ too much
3. Added get_publisher and get_subscriber factory functions. Use with 

```from msb.networking import get_publisher, get_subscriber
zmq_pub = get_publisher("zmq")
mqtt_sub = get_subscriber("mqtt", "/sensor/temp")```

4. msb/mqtt and msb/zmq have NOT been deleted but are considered deprecated.
5. Added serial pub/sub draft
6. Added msb/network/packer which provides a common interface for serializing and deserializing. Not fully tested.
7. Started work on unittests with Mock objects.

### CODE REVIEW

File: msb/network/pubsub/factories.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
+import os
+
+from .types import Publisher, Subscriber
```
Comment (k323r) [inline msb/network/pubsub/factories.py:3]: nice pattern, had to read up on it, learned a ton :+1:

**Review by k323r — COMMENTED** [review]
Impressive rewrite! No changes requested to the core code from my side, however, before merging, I'd like to make sure that the legacy publisher/subscriber model as currently used in our services still works.

File: msb/network/serial/publisher.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        self.config = config
+        self.packer = serial_packer
+        self.connect()
+
+    def connect(self):
+        self.serial: serial.Serial = serial.Serial(
+            port=self.config.port,
+            baudrate=self.config.baudrate,
+            bytesize=self.config.bytesize,
+            parity=serial.PARITY_NONE,
```
Comment (kawindme) [inline msb/network/serial/publisher.py:23]: Parity is defined in SerialConf but not used here.

Comment (pacjac) [reply msb/network/serial/publisher.py:23]: Removed from config.

**Comment by pacjac** [comment]
Addressed comments. Added functionality to import types from msb.network

e.g.

```
from msb.network import Subscriber
from msb.network import ZMQ_Publisher
from msb.network import MQTT_Subscriber
```

**Review by k323r — APPROVED** [review]

**Pull request merged**